core/rawdb/chain_freezer.go

```diff
@@ -77,6 +73,54 @@ func (f *chainFreezer) Close() error {
 	return f.Freezer.Close()
 }
 
+// block tags
+const (
+	finalizedBlock = "finalized"
+	headBlock      = "head"
+)
```

Comment:
I really dislike this string solution. MY 2c is to just have two methods: `readHeadNumer` and `readFinalizedNumber`. The code duplication really is small and it's a lot cleaner than a string switch.